src/api/app/models/notification.rb

```diff
@@ -17,6 +17,10 @@ def event
     @event ||= event_type.constantize.new(event_payload)
   end
 
+  def self.cleanup
+    NotificationsFinder.new.stale.delete_all
```

Comment:
It's not a blocker, but I would put this directly in the cleanup job. I think it's unnecessary to have a class method for this. It would remind easy to test in the specs.

Reply:
Yes, much better. I removed the `self.cleanup` method, replaced the previous test with a job test, and also added the `Job` suffix. Please have another look.